assets/src/dashboard/app/views/editorSettings/publisherLogo/gridItem.js

```diff
@@ -0,0 +1,142 @@
+import PopoverLogoContextMenu from './popoverLogoContextMenu';
+
+export const GridItem = forwardRef(
+  (
+    {
+      contextMenuId,
+      index,
+      isActive,
+      onMenuItemToggle,
+      onRemoveLogo,
+      onUpdateDefaultLogo,
+      publisherLogo,
+      setActivePublisherLogoId,
+      showLogoContextMenu,
+    },
+    ref
+  ) => {
+    const items = useMemo(
+      () => [
+        {
+          label: __('Set as Default', 'web-stories'),
+          onClick: (ev) => {
+            ev.preventDefault();
+            onUpdateDefaultLogo(publisherLogo);
+          },
+          disabled: publisherLogo.isDefault,
+        },
+        {
+          label: __('Delete', 'web-stories'),
+          onClick: (ev) => {
+            ev.preventDefault();
+            onRemoveLogo(publisherLogo, index);
+          },
+        },
+      ],
+      [index, publisherLogo, onRemoveLogo, onUpdateDefaultLogo]
+    );
+
+    if (!publisherLogo) {
+      return null;
+    }
+
+    return (
+      <GridItemContainer
+        ref={ref}
+        role="listitem"
+        active={publisherLogo.isDefault}
+        data-testid={`publisher-logo-${index}`}
+      >
+        <GridItemButton
+          onFocus={() => {
+            setActivePublisherLogoId(publisherLogo.id);
+          }}
+          data-testid={`uploaded-publisher-logo-${index}`}
+          isSelected={isActive}
+          tabIndex={isActive ? 0 : -1}
+          onClick={(e) => {
+            e.preventDefault();
+            setActivePublisherLogoId(publisherLogo.id);
+          }}
+          aria-label={sprintf(
+            /* translators: %s: logo title.*/
+            __('Publisher Logo %s', 'web-stories'),
+            publisherLogo.title
+          )}
+        >
+          <Logo src={publisherLogo.src} alt={publisherLogo.title} />
+        </GridItemButton>
+        {publisherLogo.isDefault && (
+          <CenterMutedText
+            size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL}
+          >
+            {__('Default', 'web-stories')}
+          </CenterMutedText>
+        )}
+        {showLogoContextMenu && (
+          <PopoverLogoContextMenu
+            isActive={isActive}
+            idx={index}
+            items={items}
+            publisherLogo={publisherLogo}
+            onMenuItemToggle={onMenuItemToggle}
+            contextMenuId={contextMenuId}
+          />
+        )}
+      </GridItemContainer>
+    );
+  }
+);
+export const GridItemPropTypes = {
+  contextMenuId: PropTypes.shape({
+    set: PropTypes.func.isRequired,
+    value: PropTypes.number,
+  }),
+  index: PropTypes.number.isRequired,
+  isActive: PropTypes.bool,
+  onMenuItemToggle: PropTypes.func.isRequired,
+  onRemoveLogo: PropTypes.func.isRequired,
+  onUpdateDefaultLogo: PropTypes.func.isRequired,
+  publisherLogo: PropTypes.shape({
+    src: PropTypes.string,
+    title: PropTypes.string,
+    id: PropTypes.number,
+    isDefault: PropTypes.bool,
+  }).isRequired,
```

Comment:
I think this is necessary here since this is the `GridItem` component 😅. However, I do have this defined in `assets/src/dashboard/app/views/editorSettings/publisherLogo/popoverLogoContextMenu.js` so I'll just make sure it's defined in that file only.